### PR TITLE
Added Azure Policies to Enforce NSG and UDR on all subnets

### PR DESCRIPTION
**Pull request by guidovbrakel**
Added Azure Policy to Enforce NSG on all subnets

### CODE REVIEW

File: samples/Network/enforce-nsg-on-all-subnets/azurepolicy.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            "if": {
+                "anyOf": [
+                    {
+                        "allOf": [
+                            {
+                                "field": "type",
+                                "equals": "Microsoft.Network/virtualNetworks"
+                            },
+                            {
+                                "not": {
```
Comment (robga) [inline samples/Network/enforce-nsg-on-all-subnets/azurepolicy.json:16]: Could you remove "not" and change condition to "exists": "false"

File: samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+					{
+						"allOf": [
+							{
+								"field": "type",
+								"equals": "Microsoft.Network/virtualNetworks"
+							},
+							{
+								"not": {
+									"field": "Microsoft.Network/virtualNetworks/subnets[*].routeTable",
+									"exists": true
```
Comment (robga) [inline samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json:18]: Could you remove "not" and change condition to "exists": "false"

File: samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+    "properties": {
+        "displayName": "Enforces an UDR on all subnets",
+        "description": "Denies the creation of subnets without an UDR",
+        "mode": "All",
```
Comment (robga) [inline samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json:5]: The "indexed" mode should be good for the policy rule

File: samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		"policyRule": {
+			"if": {
```
Comment (robga) [inline samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json:7]: Could you replace "tab" with 4 spaces.

**Review by robga — CHANGES_REQUESTED** [review]

File: samples/Network/enforce-nsg-on-all-subnets/azurepolicy.json (hunk 1 above)
Comment (guidovbrakel) [reply samples/Network/enforce-nsg-on-all-subnets/azurepolicy.json]: Done

File: samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json (hunk 4 above)
Comment (guidovbrakel) [reply samples/Network/enforce-route-table-on-all-subnets/azurepolicy.json:7]: Done

**Review by guidovbrakel — COMMENTED** [review]
Done

**Review by guidovbrakel — COMMENTED** [review]
Done

**Review by guidovbrakel — COMMENTED** [review]
.

**Pull request closed without merging**